java/runtime-native/src/main/java/org/ray/spi/impl/RayletStateStoreProxyImpl.java

```diff
@@ -0,0 +1,109 @@
+package org.ray.spi.impl;
+
+import com.google.flatbuffers.FlatBufferBuilder;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.ray.api.UniqueID;
+import org.ray.format.gcs.ClientTableData;
+import org.ray.format.gcs.GcsTableEntry;
+import org.ray.format.gcs.TablePrefix;
+import org.ray.spi.KeyValueStoreLink;
+import org.ray.spi.StateStoreProxy;
+import org.ray.spi.model.AddressInfo;
+import org.ray.util.NetworkUtil;
+import org.ray.util.logger.RayLog;
+import redis.clients.util.SafeEncoder;
+
+/**
+ * A class used to interface with the Ray control state for raylet.
+ */
+public class RayletStateStoreProxyImpl extends StateStoreProxyImpl {
```

Comment:
This class seems to have a lot of duplicate code with `StateStoreProxyImpl`, it'd be better to avoid this. we can either
1) put common code in a super class, or
2) re-use `StateStoreProxyImpl` and add a `useRaylet` flag to it.

Reply:
Great suggestion.